src/context.rs

```diff
@@ -974,25 +997,31 @@ impl SeafowlContext for DefaultSeafowlContext {
                     // Scan through the original table (with selection) and:
                     // SELECT [for each col, "col AS col" if not an assignment, otherwise "expr AS col"]
                     //   FROM original_table WHERE [selection]
-                    // Somehow also split the result by existing partition boundaries and leave unchanged partitions alone
+                    // Also split the result by existing partition boundaries and leave unchanged partitions alone
 
-                    // TODO we need to load the table object here in order to validate the UPDATE clauses
-                    let table_schema: DFSchema = DFSchema::empty();
+                    // Get the actual table schema, since DF needs to validate unqualified columns
+                    // (i.e. ones referenced only by column name, lacking the relation name)
+                    let table_name = name.to_string();
+                    let seafowl_table = self.try_get_seafowl_table(&table_name)?;
+                    let table_schema = seafowl_table.schema.arrow_schema.clone().to_dfschema()?;
 
                     let selection_expr = match selection {
                         None => None,
                         Some(expr) => Some(query_planner.sql_to_rex(expr, &table_schema, &mut HashMap::new())?),
                     };
 
-                    let assignments_expr = assignments.iter().map(|a| {
-                        Ok(Assignment { column: compound_identifier_to_column(&a.id)?, expr: query_planner.sql_to_rex(a.value.clone(), &table_schema, &mut HashMap::new())? })
-                    }).collect::<Result<Vec<Assignment>>>()?;
+                    let assignment_exprs = assignments.iter().map(|a| {
+                        Ok((
+                            table_schema.field_with_unqualified_name(&normalize_ident(&a.id[0]))?.name().clone(),
```

Comment:
I opted for interpreting the idents as single column names, and not as compund identifiers, as it made more sense for an `UPDATE` statement.

Reply:
That makes sense, compound identifiers are only useful in an `UPDATE ... USING` (where you might update with some values derived from a different table)